runtime/Java/src/org/antlr/v4/runtime/atn/PredictionContext.java

```diff
@@ -67,19 +63,19 @@ protected PredictionContext(int cachedHashCode) {
 	}
 
 	/** Convert a {@link RuleContext} tree to a {@link PredictionContext} graph.
-	 *  Return {@link #EMPTY} if {@code outerContext} is empty or null.
+	 *  Return {@link EmptyPredictionContext#Instance} if {@code outerContext} is empty or null.
 	 */
 	public static PredictionContext fromRuleContext(ATN atn, RuleContext outerContext) {
-		if ( outerContext==null ) outerContext = RuleContext.EMPTY;
+		if ( outerContext==null ) outerContext = ParserRuleContext.EMPTY;
```

Comment:
Are we sure this is always ParserRuleContext and not LexerRuleContext?

Reply:
haha. there is no LexerRuleContext.  trying read this again after 10 years

Reply:
actually, RuleContext might be what we need as an empty for it is general for lexer and parser I think. Why did you convert to ParserRuleContext?

Reply:
I didn't change the type, it always was `ParserRuleContext`: https://github.com/antlr/antlr4/pull/3752/commits/4b6918a52e7959935ee128b05275a6fad01cbf41#diff-20c6552f826df3e37bfcf31155a2e4360d2b4fea4fbb4ad6d27832946ff7718fL69 Also, there is no class named `LexerRuleContext` in ANTLR runtime library.

Reply:
ah!! cool.